src/ssd1306.toit

```diff
@@ -92,14 +96,16 @@ class SSD1306 extends AbstractDriver:
     command_ top >> 3           // Page start.
     command_ (bottom >> 3) - 1  // Page end.
 
-    width := right - left
+    patch_width := right - left
+
+    line_buffer := buffer_[0..patch_width + 1]
 
-    buffer := ByteArray width + 1
-    buffer[0] = 0x40
+    line_buffer[0] = 0x40
 
     i := 0
-    ((bottom - top) >> 3).repeat:
-      width.repeat: buffer[it + 1] = pixels[i++]
-      i2c_.write buffer: print "error printing data"
+    for y := top; y < bottom; y += 8:
+      line_buffer.replace 1 pixels i i + patch_width
+      i += patch_width
+      i2c_.write line_buffer: print "error printing data"
```

Comment:
Looks like a weird pseudo-error. Is it a warning? Is it an error? Should it be an exception?